dftimewolf/lib/args_validator.py

```diff
@@ -0,0 +1,294 @@
+"""Validators for recipe arguments."""
+
+import abc
+import ipaddress
+import re
+
+from typing import Any, Dict, Optional
+
+from dftimewolf.lib import errors
+
+
+class AbstractValidator(abc.ABC):
+  """Base class for validator objects."""
+
+  VALIDATOR_NAME: str = None  # type: ignore
```

Comment:
nit: We can probable call these NAME, since, we're doing things like validator.VALIDATOR_NAME

Reply:
Done.